core/src/main/scala/cats/Foldable.scala

```diff
@@ -152,6 +152,26 @@ import simulacrum.typeclass
   def size[A](fa: F[A]): Long = foldMap(fa)(_ => 1)
 
   /**
+    * Get the element at the index of the `Foldable`.
+    *
+    * The default implementation of this is based on `foldLeft`, and thus will
+    * always fold across the entire structure.
+    */
+  def get[A](fa: F[A])(idx: Long): Option[A] =
+    foldLeft(fa, (0L, Option.empty[A])) { (b, a) =>
+      b._1 match {
+        case i if i == idx  => (i + 1, Some(a))
+        case i if i < idx => (i + 1, None)
+        case _ => b
+      }
+    }._2
+
+  /**
+    * Get the element at the index of the `Foldable`
+    */
+  def apply[A](fa: F[A])(idx: Long): A = get(fa)(idx).get
```

Comment:
@jedws Can you point to specific ones in Cats? I would guess `Try` is in that set, which is sketchy in that it's existence was motivated by using `Throwable`, but if users just treat it as a glorified `A => Either[Throwable, A]` I think it is lawful. It certainly breaks in the presence of exceptions in that `Try`'s constructor captures those, but the counter-argument to that is at the point where you're throwing then all bets are off anyways. I agree though that the instance is sketchy.

I did a search for `unsafe` in the codebase and the only ones I see are on the `NonEmpty` stuff that lets you index into a `NonEmptyVector` unsafely. While I personally don't like it, it may there for performance reasons, maybe?

On a related note, for stuff like `Functor[Set]` we kick those out to a [separate lib](https://github.com/non/alleycats) for those who want to make that gamble.